docker-compose.image.example.yaml

```diff
@@ -1,75 +1,208 @@
+      - GITHUB_USERNAME=$GITHUB_USERNAME
+      - GITHUB_ACCESS_TOKEN=$GITHUB_ACCESS_TOKEN
+      - JIRA_INSTANCE_URL=$JIRA_INSTANCE_URL
+      - JIRA_USERNAME=$JIRA_USERNAME
+      - JIRA_API_TOKEN=$JIRA_API_TOKEN
+      - SLACK_BOT_TOKEN=$SLACK_BOT_TOKEN
+      - STABILITY_API_KEY=$STABILITY_API_KEY
+      - ENGINE_ID=$ENGINE_ID
+      - RESOURCE_VECTOR_STORE=$RESOURCE_VECTOR_STORE
+      - RESOURCE_VECTOR_STORE_INDEX_NAME=$RESOURCE_VECTOR_STORE_INDEX_NAME
+      - REDIS_VECTOR_STORE_URL=$REDIS_VECTOR_STORE_URL
+      - QDRANT_PORT=$QDRANT_PORT
+      - QDRANT_HOST_NAME=$QDRANT_HOST_NAME
+      - CHROMA_HOST_NAME=$CHROMA_HOST_NAME
+      - CHROMA_PORT=$CHROMA_PORT
     command: ["/app/entrypoint_celery.sh"]
     volumes:
-      - "./workspace:/app/workspace"
+      - ./workspace:/app/workspace
 
   gui:
     image: "superagidev/superagi-frontend:main"
+    restart: unless-stopped
     environment:
       - NEXT_PUBLIC_API_BASE_URL=/api
-    networks:
-      - super_network
 
-  super__redis:
+  redis:
     image: "redis/redis-stack-server:latest"
-    networks:
-      - super_network
-#    uncomment to expose redis port to host
-#    ports:
-#      - "6379:6379"
+    restart: unless-stopped
+    healthcheck:
+      test: ["CMD", "redis-cli", "ping"]
+      interval: 1s
+      timeout: 3s
+      retries: 30
+    #    uncomment to expose redis port to host
+    #    ports:
+    #      - "6379:6379"
     volumes:
-      - redis_data:/data
+      - superagi_redis_data:/data
 
-  super__postgres:
+  postgres:
     image: "docker.io/library/postgres:latest"
+    restart: unless-stopped
     environment:
-      - POSTGRES_USER=superagi
-      - POSTGRES_PASSWORD=password
-      - POSTGRES_DB=super_agi_main
+      - POSTGRES_USER=$DB_USERNAME
+      - POSTGRES_PASSWORD=$DB_PASSWORD
+      - POSTGRES_DB=$DB_NAME
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready", "-d", "$DB_NAME"]
+      interval: 5s
+      timeout: 60s
+      retries: 10
+      start_period: 10s
     volumes:
       - superagi_postgres_data:/var/lib/postgresql/data/
-    networks:
-      - super_network
-#    uncomment to expose postgres port to host
-#    ports:
-#      - "5432:5432"
+  #    uncomment to expose postgres port to host
+  #    ports:
+  #      - "5432:5432"
+
+  weaviate:
+    image: semitechnologies/weaviate:1.21.1
+    restart: unless-stopped
+    volumes:
+      - superagi_weaviate_data:/var/lib/weaviate
+    command:
+      - --host
+      - 0.0.0.0
+      - --port
+      - "8080"
+      - --scheme
+      - http
+    environment:
+      PERSISTENCE_DATA_PATH: "/var/lib/weaviate"
+      AUTHENTICATION_APIKEY_ENABLED: true
+      AUTHENTICATION_APIKEY_ALLOWED_KEYS: $WEAVIATE_APIKEY
+      AUTHENTICATION_APIKEY_USERS: $WEAVIATE_APIKEY
+      ENABLE_MODULES: text2vec-openai,generative-openai
 
   proxy:
     image: nginx:stable-alpine
+    restart: unless-stopped
     ports:
-      - "3000:80"
-    networks:
-      - super_network
+      - 7896:80
```

Comment:
any specific reason for choosing 7896

Reply:
Not really, just a random one.

Reply:
so let make it 3000
and add in docs that it can be changed here

Reply:
I'd go with an env var then, since 3000 is used by a lot of node/rails applications which would collide.